parquet/src/main/java/org/apache/iceberg/parquet/ParquetSchemaUtil.java

```diff
@@ -82,23 +86,34 @@ public static MessageType pruneColumnsFallback(MessageType fileSchema, Schema ex
     return builder.named(fileSchema.getName());
   }
 
-  public static boolean hasIds(MessageType fileSchema) {
-    try {
-      // Try to convert the type to Iceberg. If an ID assignment is needed, return false.
-      ParquetTypeVisitor.visit(fileSchema, new MessageTypeToType(fileSchema) {
-        @Override
-        protected int nextId() {
-          throw new IllegalStateException("Needed to assign ID");
-        }
-      });
+  /**
+   * Prunes columns from a Parquet file schema that was written without field ids.
+   * The order of columns in the resulting Parquet schema matches the Parquet file.
+   *
+   * @param fileSchema schema from a Parquet file that does not have field ids.
+   * @param expectedSchema expected schema
+   * @return a parquet schema pruned using the expected schema
+   */
+  public static MessageType pruneColumnsByName(MessageType fileSchema, Schema expectedSchema, NameMapping nameMapping) {
```

Comment:
We don't need this method. Instead, let's add the IDs and then use normal schema pruning.

Reply:
Agreed.